File: tournaments/isMonotonous/isMonotonous.js
function isMonotonous(sequence) {
    const n = sequence.length;

    if (1 === n) return true;
    
    const a = sequence[1] - sequence[0];
    
    for (let i = 0; n - 1 > i; ++i)
        if (0 >= a * (sequence[i + 1] - sequence[i]))
            return false;

    return true;
}

File: tournaments/isMonotonous/isMonotonous.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub isMonotonous {
    my ($sequence) = @_;

    my $n = @$sequence;

    if (1 == $n) {
        return 1;
    }

    my $diff = $$sequence[1] - $$sequence[0];

    for (my $i = 0; $n - 1 > $i; ++$i) {
        if (0 >= $diff * ($$sequence[$i + 1] - $$sequence[$i])) {
            return 0;
        }
    }

    return 1;
}
